File: Habitica/src/main/java/com/habitrpg/android/habitica/models/social/ChatMessageLike.java
package com.habitrpg.android.habitica.models.social;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by phillip on 30.06.17.
 */

public class ChatMessageLike extends RealmObject {

    @PrimaryKey
    public String id;

    public ChatMessageLike(String id) {
        super();
        this.id = id;
    }

    public ChatMessageLike() {
        super();
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/social/ChatMessageLike.kt
package com.habitrpg.android.habitica.models.social

import com.habitrpg.android.habitica.models.BaseObject
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass

@RealmClass(embedded = true)
open class ChatMessageLike : RealmObject, BaseObject {

    @PrimaryKey
    var key: String = ""

    var messageId: String = ""
        set(value) {
            field = value
            key = messageId + id
        }
    var id: String = ""
        set(value) {
            field = value
            key = messageId + id
        }
    constructor(id: String, messageId: String) : super() {
        this.id = id
        this.key = messageId + id
    }

    constructor() : super()
}
